src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -9,140 +9,149 @@
+			throw new InvalidStorageFilePathException("Storage file should end with '.txt'");
+		}
+	}
+
+	/**
+	 * Returns true if the given path is acceptable as a storage file.
+	 * The file path is considered acceptable if it ends with '.txt'
+	 */
+	private static boolean isValidPath(Path filePath) {
+		return filePath.toString().endsWith(".txt");
+	}
+
+	/**
+	 * Saves all data to this storage file.
+	 *
+	 * @throws StorageOperationException if there were errors converting and/or storing data to file.
+	 */
+	public void save(AddressBook addressBook) throws StorageOperationException {
+
+		/* Note: Note the 'try with resource' statement below.
+		 * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
+		 */
+		try (final Writer fileWriter =
+				new BufferedWriter(new FileWriter(path.toFile()))) {
+
+			final AdaptedAddressBook toSave = new AdaptedAddressBook(addressBook);
+			final Marshaller marshaller = jaxbContext.createMarshaller();
+			marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+			marshaller.marshal(toSave, fileWriter);
+
+		} catch (IOException ioe) {
+			throw new StorageOperationException("Error writing to file: " + path);
+		} catch (JAXBException jaxbe) {
+			throw new StorageOperationException("Error converting address book into storage format");
+		}
+	}
+
+	/**
+	 * Loads data from this storage file.
+	 *
+	 * @throws StorageOperationException if there were errors reading and/or converting data from file.
+	 */
+	public AddressBook load() throws StorageOperationException {
+		try (final Reader fileReader =
+				new BufferedReader(new FileReader(path.toFile()))) {
+
+			final Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+			final AdaptedAddressBook loaded = (AdaptedAddressBook) unmarshaller.unmarshal(fileReader);
+			// manual check for missing elements
+			if (loaded.isAnyRequiredFieldMissing()) {
+				throw new StorageOperationException("File data missing some elements");
+			}
+			return loaded.toModelType();
+
+			/* Note: Here, we are using an exception to create the file if it is missing. However, we should minimize
+			 * using exceptions to facilitate normal paths of execution. If we consider the missing file as a 'normal'
+			 * situation (i.e. not truly exceptional) we should not use an exception to handle it.
+			 */
+
+			// create empty file if not found
+		} catch (FileNotFoundException fnfe) {
+			final AddressBook empty = new AddressBook();
+			save(empty);
+			return empty;
+
+			// other errors
+		} catch (IOException ioe) {
+			throw new StorageOperationException("Error writing to file: " + path);
+		} catch (JAXBException jaxbe) {
+			throw new StorageOperationException("Error parsing file data format");
+		} catch (IllegalValueException ive) {
+			throw new StorageOperationException("File contains illegal data values; data type constraints not met");
+		}
+	}
+
+	public String getPath() {
+		return path.toString();
+	}
```

Comment:
Avoid auto-formatting unrelated parts of the code. Doing so can result in unrelated changes being included in the commit. In Eclipse, select the code block you want to auto-format before using the auto-format feature. Otherwise the whole file get auto-formatted. You can use sourcetree to look through changes before committing, and discard any unrelated changes.